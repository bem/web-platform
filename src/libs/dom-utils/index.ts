export * from './canUseDom'
export * from './focus'
export * from './input-cursor'
export * from './is-hidden'
