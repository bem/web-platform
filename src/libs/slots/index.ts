export * from './useSlots'
export * from './createSlot'
export * from './types'
