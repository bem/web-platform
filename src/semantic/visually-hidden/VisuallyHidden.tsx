import React, { CSSProperties, HTMLAttributes, PropsWithChildren, forwardRef } from 'react'

const VisuallyHiddenStyles: CSSProperties = {
  border: 0,
  clip: 'rect(0 0 0 0)',
  clipPath: 'inset(50%)',
  height: 1,
  margin: '0 -1px -1px 0',
  overflow: 'hidden',
  padding: 0,
  position: 'absolute',
  width: 1,
  whiteSpace: 'nowrap',
}

export type VisuallyHiddenProps = HTMLAttributes<HTMLSpanElement>

export const VisuallyHidden = forwardRef<HTMLSpanElement, PropsWithChildren<VisuallyHiddenProps>>(
  (props, ref) => {
    const { children, ...otherProps } = props

    return (
      <span ref={ref} style={VisuallyHiddenStyles} {...otherProps}>
        {children}
      </span>
    )
  },
)
