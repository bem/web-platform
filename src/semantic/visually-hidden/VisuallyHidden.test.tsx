import { createClientRender, screen } from '../../libs/testing'
import { VisuallyHidden } from './VisuallyHidden'

const hiddenText = 'Hidden text'

describe('useFocusTrap', () => {
  const render = createClientRender()

  test('should render element with proper styles', () => {
    render(<VisuallyHidden>{hiddenText}</VisuallyHidden>)
    expect(screen.getByText('Hidden text')).toHaveStyle({
      border: 0,
      clip: 'rect(0 0 0 0)',
      clipPath: 'inset(50%)',
      height: '1px',
      margin: '0 -1px -1px 0',
      overflow: 'hidden',
      padding: 0,
      position: 'absolute',
      width: '1px',
      whiteSpace: 'nowrap',
    })
  })

  test('should match snapshot', () => {
    render(
      <button data-testid="test">
        <VisuallyHidden>{hiddenText}</VisuallyHidden>
      </button>,
    )
    expect(screen.getByTestId('test')).toMatchSnapshot()
  })
})
