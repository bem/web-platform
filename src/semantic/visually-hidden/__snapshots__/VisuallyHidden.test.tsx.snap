// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFocusTrap should match snapshot 1`] = `
<button
  data-testid="test"
>
  <span
    style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
  >
    Hidden text
  </span>
</button>
`;
